app/scripts/ViewConfigEditor.js

```diff
@@ -0,0 +1,160 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Editor from 'react-simple-code-editor';
+import { highlight, languages } from 'prismjs/components/prism-core';
+import 'prismjs/components/prism-json';
+
+import Button from './Button';
+import Dialog from './Dialog';
+import withModal from './hocs/with-modal';
+import withPubSub from './hocs/with-pub-sub';
+import { timeout } from './utils';
+
+import '../styles/ViewConfigEditor.module.scss';
+
+class ViewConfigEditor extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      code: props.viewConfig,
+      hide: false,
+    };
+
+    this.handleChangeBound = this.handleChange.bind(this);
+    this.handleKeyDownBound = this.handleKeyDown.bind(this);
+    this.handleKeyUpBound = this.handleKeyUp.bind(this);
+    this.hideBound = this.hide.bind(this);
+    this.showBound = this.show.bind(this);
+
+    this.pubSubs = [];
+
+    this.pubSubs.push(
+      this.props.pubSub.subscribe('keydown', this.handleKeyDownBound),
+    );
+    this.pubSubs.push(
+      this.props.pubSub.subscribe('keyup', this.handleKeyUpBound),
+    );
+  }
+
+  async componentDidMount() {
+    if (this.editor) {
+      this.editor._input.focus();
+      this.editor._input.setSelectionRange(0, 0);
+      await timeout(0);
```

Comment:
What's this for?

Reply:
This autofocuses the textarea and scrolls to the top. Apparently when the content in the textarea is large and the textarea is focused Chrome will automatically scroll to the end of it, which is a little confusing when you first load the view config.